### ISSUE TITLE
return type of `incident` varies with indexing

### ISSUE THREAD

**Issue by slwu89** [AlgebraicJulia/Catlab.jl#597]
Hi,

I noticed that depending if an `Attr` is indexed or not will affect what `incident` returns. A minimal example is below:

I'm using Catlab version 0.13.7.

```julia
using Catlab
using Catlab.CategoricalAlgebra
using Catlab.CategoricalAlgebra.FinSets
using Catlab.Present
using Catlab.Theories

@present MySchema(FreeSchema) begin
    State::Ob
    StateLabel::AttrType
    statelabel::Attr(State, StateLabel)
end

@acset_type MySchema_index(MySchema,unique_index = [:statelabel])
@acset_type MySchema_noindex(MySchema)

MySchema_index_obj = MySchema_index{String}()
MySchema_noindex_obj = MySchema_noindex{String}()

add_parts!(MySchema_index_obj, :State, 3, statelabel = ["A", "B", "C"])
add_parts!(MySchema_noindex_obj, :State, 3, statelabel = ["A", "B", "C"])

# returns scalar
incident(MySchema_index_obj, "C", :statelabel)

# returns vector
incident(MySchema_noindex_obj, "C", :statelabel)
```

**Comment by epatters** [comment]
On the one hand, this is "intended behavior": it has been the behavior of unique indexing since the original implementation of acsets, and we have preserved it through the various refactorings. Changing it would be a breaking change.

OTOH, I think it was a design mistake because indexing decisions, including unique indexing, should not affect the semantics of the acset. What should happen here is that `incident` returns a `SVector{1,T}` for unique indexed morphisms/attributes. That would be just as performant but would preserve the vector semantics of incident.

The origin of the issue is that `unique_index` was used as a quick-and-dirty way to handle monic constraints, which currently cannot be expressed in the schema.

TLDR; fixing this would be a breaking change, but at some point, possibly the next breaking release, we should do it.

**Comment by slwu89** [comment]
Thanks for the explanation @epatters, that all makes sense.
